notes/lecture22.tex

```diff
@@ -0,0 +1,205 @@
+Sparse linear regression has received a lot of attention and there are many related optimization programs like:
+BPDN (allows for noise)
+$$\min_{\|Ax-y\|_2 \leq s} \|x\|_1$$
+
+Constrained Lasso
+$$\min_{\|x\|_1 \leq \lambda }\|Ax-y\|_2^2 $$
+
+Lagrangian Lasso
+$$\min \|Ax-y\|_2^2 +  \lambda \|x\|_1$$
+
+In our setting, we have \\
+
+\centering{nonconvex constraint} 
+$$\downarrow$$
+\centering{cvx relaxation }
+$$\downarrow$$
+\centering{PGD (Projected Gradient Descent)},
+
+or more directly, nonconvex constraint $\rightarrow$ PGD. 
+
+Here, the Restricted Isometry Property allows optimization over nonconvex sets.
+
+
+Now, we discuss the Iterative Hard Thresholding (IHT) which is known as a projected gradient descent (PGD) for sparse vectors.
+
+Iterative Hard Thresholding (IHT) is used with Gradient Descent for the following problem,
+
+\begin{align}
+f(x) &= \frac{1}{2} \| Ax - y \|_2^2 \\
+\nabla f(x) &= A^{\trans}(Ax - y)
+\end{align}
+
+If it is possible to directly optimize over nonconvex sets, then we don't need convex relaxation.
+
+We want to show that the algorithm IHT outputs a solution. The algorithm is defined as:
+
+\begin{align*}
+\text{IHT}(y, A, t, s) \\
+x^1 &\leftarrow 0 \\
+\text{ for } &i=1 ,\cdots, t, \\
+\tilde{x}^{i+1} &\leftarrow x^i - A^{\trans} (Ax - y) \\
+x^{i+1} &\leftarrow P_S( \tilde{x}^{i+1} ) \\
+\text{ return }& \hat{x} \leftarrow x^{t+1}
+\end{align*}
+
+Note that $P_s$ is a projection onto a set of s-sparse vectors. In the rest of the lecture we will see how we should do this projection and how fast the method is.
+
+RIP: matrix $A$ satisfies the $(s,\delta)$-RIP if for all $s$-sparse vectors we have:
+$$(1-\delta) \|x\|_2^2 \leq \|Ax\|_2^2 \leq (1+\delta)\|x\|_2^2$$
+
+Consequences: For all supports $S$ of size $s$, we have:
+$$\|I-X_s^{\trans} X_s\|_2 \leq \delta$$
+
+Setup: \\
+$y = Ax^* + e$ \\
+$x^*$ is $s$-sparse with support $S^*$ \\ 
+$A$ has ($3s,\frac{1}{4})$-RIP \\
+$e$ is arbitrary noise.
+Then the following holds,
+\begin{theorem}
+$$\|x^{i+1}-x^*\|_2 \leq \frac{1}{2} \|x^{i}-x^*\|_2 + \max_{|S| \leq 3s} \|A_s^{\trans} e\|_2$$
+\end{theorem}
+
+\begin{proof}
+Let $S_i = Supp. (x^i)$ and $ S\prime = S^{i+1} \cup S^i \cup S^*$ (so $|S \prime \leq 3s|$). Then:
+
+\begin{align*}
+\|x^{i+1}-x^*\|_2 &\leq \|x^{i+1} - \tilde x_{\prime S}^{i+1}\|_2 + \|\tilde x_{\prime S}^{i+1} -x^*\|_2\  \tag{ by "triangle inequality"}\\
+			 &\leq 2 \|\tilde x_{ S\prime}^{i+1} -x^*\|_2\\ 
+			 &= 2 \|x_{ S\prime}^i - A_ { S\prime}^{\trans}(Ax^i -y)-x^*\|\\
+			 &= 2 \|x^i - A_ {S\prime}^{\trans}(A_ { S \prime}x^i -A_ { S\prime}x^*-e)-x^*\|\\
+			 &= 2 \|x^i - x^*- A_ { S\prime}^{\trans}A_ { S\prime} (x^i-x^*) + A_ {S \prime}^{\trans} e\|\\
+			 &\leq 2\| I -  A_ {S \prime}^{\trans} A_ {S \prime} (x^i-x^*)\|_2 +   2\| A_ {S \prime}^{\trans} e\|_2 \\
+			 &\leq 2 \delta \| (x^i-x^*)\|_2 +   2 \max_{|S| \leq 3s}\| A_ {S}^{\trans} e\|_2
+\end{align*}
+\end{proof}
+
```

Comment:
A quick transition here could be helpful for the reader.

Reply:
We are not completely sure what could be a good transition here. Is it enough to say:

"In the following part, we show the rate of convergence of projected gradient descent for the above mentioned setup."

If you have better words for that to make more clarification for this part, please let us to know.

Reply:
I was thinking about something along the lines of

"In the theorem above, we have shown that the error goes down by a factor 1/2 in every iteration (up to the noise threshold). From there, it is fairly straightforward to get a linear convergence rate as in the following corollary."

(maybe a bit more formal)

Basically we want to tell the reader how the theorem above and the following corollary fit together.

Reply:
Got it, thanks.